packages/pigeon/lib/functional.dart

```diff
@@ -0,0 +1,48 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// A [map] function that calls the function with an enumeration.
+Iterable<U> intMap<T, U>(
+    Iterable<T> iterable, U Function(int x, T value) func) sync* {
+  int count = 0;
+  for (final T value in iterable) {
+    yield func(count, value);
+    count += 1;
+  }
+}
+
+/// Performs like [forEach] with an enumeration.
```

Comment:
Again, it seems like there's a mismatch here between the description—forEach already works on enumerations, because `Iterable` is an enumeration`—and the functionality, which seems to be index-based as above.

Reply:
file removed